osu.Game/Overlays/Toolbar/Toolbar.cs

```diff
@@ -120,9 +80,69 @@ public Toolbar()
             };
 
             RelativeSizeAxes = Axes.X;
-            Size = new Vector2(1, height);
+            Size = new Vector2(1, HEIGHT);
+        }
+
+        public class ToolbarBackground : Container
+        {
+            private Box solidBackground;
+            private Box gradientBackground;
+
+            public ToolbarBackground()
+            {
+                RelativeSizeAxes = Axes.Both;
+                Children = new Drawable[]
+                {
+                    solidBackground = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = OsuColour.Gray(0.1f),
+                        Alpha = alpha_normal,
+                    },
+                    gradientBackground = new Box
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Anchor = Anchor.BottomLeft,
+                        Alpha = 0,
+                        Height = 90,
+                        ColourInfo = ColourInfo.GradientVertical(
+                            OsuColour.Gray(0.1f).Opacity(0.5f), OsuColour.Gray(0.1f).Opacity(0)),
+                    },
+                };
+            }
+
+            protected override bool OnHover(InputState state)
+            {
+                solidBackground.FadeTo(alpha_hovering, transition_time, EasingTypes.OutQuint);
+                gradientBackground.FadeIn(transition_time, EasingTypes.OutQuint);
+                return true;
+            }
+
+            protected override void OnHoverLost(InputState state)
+            {
+                solidBackground.FadeTo(alpha_normal, transition_time, EasingTypes.OutQuint);
+                gradientBackground.FadeOut(transition_time, EasingTypes.OutQuint);
+            }
         }
 
         public void SetGameMode(PlayMode mode) => modeSelector.SetGameMode(mode);
+
+        protected override void PopIn()
+        {
+            MoveToY(0, transition_time, EasingTypes.OutQuint);
+            FadeIn(transition_time, EasingTypes.OutQuint);
+        }
+
+        protected override void PopOut()
+        {
+            MoveToY(-DrawSize.Y, transition_time, EasingTypes.InQuint);
+            FadeOut(transition_time, EasingTypes.InQuint);
+        }
+
+        class PassThroughFlowContainer : FlowContainer
+        {
+            //needed to get input to the login overlay.
+            public override bool Contains(Vector2 screenSpacePos) => true;

```

Comment:
I would pass-through the way it is done in ToolbarUserButton, rather than always returning true.